core/src/test/java/org/apache/calcite/test/TopDownOptTest.java

```diff
@@ -64,93 +69,243 @@
  * <li>Run the test one last time; this time it should pass.
  * </ol>
  */
-class TopDownOptTest extends RelOptTestBase {
-
-  protected DiffRepository getDiffRepos() {
-    return DiffRepository.lookup(TopDownOptTest.class);
-  }
-
-  Sql sql(String sql) {
-    VolcanoPlanner planner = new VolcanoPlanner();
-    // Always use top-down optimization
-    planner.setTopDownOpt(true);
-    planner.addRelTraitDef(ConventionTraitDef.INSTANCE);
-    planner.addRelTraitDef(RelCollationTraitDef.INSTANCE);
-
-    RelOptUtil.registerDefaultRules(planner, false, false);
-
-    // Keep deterministic join order
-    planner.removeRule(JoinCommuteRule.INSTANCE);
-    planner.removeRule(JoinPushThroughJoinRule.LEFT);
-    planner.removeRule(JoinPushThroughJoinRule.RIGHT);
-
-    // Always use merge join
-    planner.removeRule(EnumerableRules.ENUMERABLE_JOIN_RULE);
-    planner.removeRule(EnumerableRules.ENUMERABLE_CALC_RULE);
-
-    // Always use sorted agg
-    planner.removeRule(EnumerableRules.ENUMERABLE_AGGREGATE_RULE);
-    planner.addRule(EnumerableRules.ENUMERABLE_SORTED_AGGREGATE_RULE);
-
-    Tester tester = createTester().withDecorrelation(true)
-        .withClusterFactory(cluster -> RelOptCluster.create(planner, cluster.getRexBuilder()));
-
-    return new Sql(tester, sql, null, planner,
-        ImmutableMap.of(), ImmutableList.of());
-  }
-
+class TopDownOptTest {
```

Comment:
Checkout this issue: CALCITE-4027
I think we still needs `extends RelOptTestBase` if we want to leverage it. Otherwise we have to manually copy from `TopDownOptTest_actual.xml`.

Reply:
Added `extends RelOptTestBase`